src/pages/User/components/UserHeader/UserHeader.vue

```diff
@@ -57,15 +63,31 @@ import userDefaultImg from 'src/assets/images/user-default.svg';
 import './styles.scss';
 import BackButton from 'src/components/BackButton/BackButton.vue';
 import QrCode from 'qrcode-vue3';
+import { handleAvatarUpload } from './utils/handleAvatarUpload';
+import { ref, Ref } from 'vue';
 
 export interface UserHeaderProps {
   isMembershipActive: boolean;
   expirationDate: Date;
   avatar?: string;
   memberCode: string;
+  newAvatar?: string;
+}
+
+interface UserHeaderEmits {
+  (eventName: 'saveAvatar', value: string): void;
 }
 
+const showAvatar: Ref<string | undefined> = ref('');
 const props = defineProps<UserHeaderProps>();
+const emits = defineEmits<UserHeaderEmits>();
+showAvatar.value = props.avatar ? props.avatar : userDefaultImg;
+const uploadImage = async (event: Event) => {
+  const file = (event.target as HTMLInputElement).files?.item(0);
+  const uploadedImage = await handleAvatarUpload(file);
+  emits('saveAvatar', uploadedImage);
+  showAvatar.value = uploadedImage;
+};
```

Comment:
Why is this event two times? Here and in `src/pages/User/components/UserHeader/utils/handleAvatarUpload.ts` file?

Both are needed? Can one of them to be removed?

Reply:
This function seems to me necessary to be able to update the api and localstorage to visualize the change.

Reply:
It's ok so, thanks for confirm!